packages/report-server/src/routes/fetchReport.ts

```diff
@@ -11,13 +11,10 @@ import { ReportBuilder } from '../reportBuilder';
 import { ReportsRequest } from '../types';
 
 class FetchReportRouteHandler {
-  private aggregator: Aggregator;
-
-  constructor() {
-    this.aggregator = createAggregator(Aggregator);
-  }
-
   fetchReport = async (req: ReportsRequest, res: Response): Promise<void> => {
+    const aggregator = createAggregator(Aggregator, {
+      session: { getAuthHeader: () => req.headers.authorization },
+    });
```

Comment:
Now need a new aggregator per request, so we can associate the user session with it.